webapp/channels/src/actions/notification_actions.jsx

```diff
@@ -111,16 +112,17 @@ export function sendDesktopNotification(post, msgProps) {
         const userStatus = getStatusForUserId(state, user.id);
         const member = getMyChannelMember(state, post.channel_id);
         const isCrtReply = isCollapsedThreadsEnabled(state) && post.root_id !== '';
+        const skipChecks = Boolean(post.props?.force_notification);
 
         if (!member) {
             return {status: 'error', reason: 'no_member'};
         }
 
-        if (isChannelMuted(member)) {
+        if (isChannelMuted(member) && !skipChecks) {
```

Comment:
We're repeating this check a LOT. Can we maybe refactor a bit, or just simply wrap the whole thing in one `if (!skipChecks)`. I feel like this way, it's way too easy to miss one.

Reply:
The refactor was heavier than I expected, but it is done. Please re-review.